OrbitGl/App.cpp

```diff
@@ -393,22 +393,14 @@ void OrbitApp::ListSessions() {
       Path::ListFiles(Path::GetPresetPath(), ".opr");
   std::vector<std::shared_ptr<Session>> sessions;
   for (std::string& filename : sessionFileNames) {
-    std::ifstream file(filename, std::ios::binary);
-    if (file.fail()) {
-      ERROR("Loading session from \"%s\": %s", filename, "file.fail()");
-      continue;
-    }
-
-    try {
-      auto session = std::make_shared<Session>();
-      cereal::BinaryInputArchive archive(file);
-      archive(*session);
-      file.close();
-      session->m_FileName = filename;
-      sessions.push_back(session);
-    } catch (std::exception& e) {
-      ERROR("Loading session from \"%s\": %s", filename, e.what());
+    auto session = std::make_shared<Session>();
+    outcome::result<void, std::string> result = 
+      ReadSessionFromFile(filename, session.get());
+    if (result.has_error()) {
+      ERROR("Loading session failed: \"%s\"", result.error());
```

Comment:
send something to ui maybe?

Reply:
I'm not sure here. This is not triggered by a user action and loads all sessions in the background. If we send every error to the UI, this could be disruptive. If this fails, the session will not show up in the SessionsDataView, so it's at least apparent to the user that they were not loaded. WDYT?

Reply:
I'll merge as-is now, but let me know if we should follow up on this.